dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -2295,4 +2298,92 @@ public Set<OrcidAuthorInformation> getWorkflowVersionOrcidAuthors(@Parameter(hid
+                .filter(w -> w.getSourceControl() == SourceControl.GITHUB)
+                .collect(Collectors.groupingBy(w -> w.getOrganization() + '/' + w.getRepository()));
+
+        final List<ZenodoHelper.GitHubRepoDois> gitHubRepoDois = repoToWorkflowsMap.keySet().stream().sorted()
+                .map(ZenodoHelper::findDoisForGitHubRepo)
+                .flatMap(Collection::stream)
+                .toList();
+
+        final Set<Workflow> updatedWorkflows = new HashSet<>();
+        gitHubRepoDois.stream().forEach(gitHubRepoDoi -> {
+            final String repo = gitHubRepoDoi.repo();
+            final String conceptDoi = gitHubRepoDoi.conceptDoi();
+            final List<Workflow> workflows = repoToWorkflowsMap.get(repo);
+            workflows.stream().forEach(workflow -> {
+                final Doi existingGitHubDoi = workflow.getConceptDois().get(Doi.DoiInitiator.GITHUB);
+                if (existingGitHubDoi != null && conceptDoi.equals(existingGitHubDoi.getName())) {
+                    LOG.warn("Skipping DOI %s for workflow %s because it already has a Zenodo DOI from GitHub".formatted(conceptDoi, workflow.getWorkflowPath()));
+                } else {
+                    final boolean noDoiToStart = workflow.getConceptDois().isEmpty();
+                    if (existingGitHubDoi == null) { // No Concept DOI yet, add it.
+                        workflow.getConceptDois().put(Doi.DoiInitiator.GITHUB,
+                                ZenodoHelper.getDoiFromDatabase(Doi.DoiType.CONCEPT, Doi.DoiInitiator.GITHUB, conceptDoi));
+                        updatedWorkflows.add(workflow);
+                    }
+                    // Add version DOI(s) to new or existing Concept DOI.
+                    gitHubRepoDoi.tagAndDoi().stream().forEach(tagAndDoi -> {
+                        final WorkflowVersion workflowVersion = workflowVersionDAO.getWorkflowVersionByWorkflowIdAndVersionName(
+                                workflow.getId(), tagAndDoi.gitHubTag());
+                        if (workflowVersion != null) {
+                            if (workflowVersion.getDois().get(Doi.DoiInitiator.GITHUB) == null) {
+                                final Doi versionDoi = ZenodoHelper.getDoiFromDatabase(Doi.DoiType.VERSION, Doi.DoiInitiator.GITHUB,
+                                        tagAndDoi.doi());
+                                workflowVersion.getDois().put(Doi.DoiInitiator.GITHUB, versionDoi);
+                                updatedWorkflows.add(workflow);
+                            }
+                        }
+                    });
+                    if (noDoiToStart) {
+                        // The default DOI selection is USER. If there was no DOI to begin with, set it to GITHUB so it will show up in the UI.
+                        workflow.setDoiSelection(Doi.DoiInitiator.GITHUB);
+                    }
+                }
+            });
+        });
```

Comment:
If we throw somewhere in the big loop (because of a coding error, connectivity problem, change in response format, etc), do we want to try to catch the exception (in some cases?) and try to keep soldiering on?